src/frontend/e2e/pages/basePage.ts

```diff
@@ -71,13 +90,30 @@ export abstract class BasePage {
   async fillByName(name: string, value: string) {
     await this.page.fill(`[name="${name}"]`, value);
   }
-
+  async fillByTypeAndName(type: string, name: string, value: string) {
+    await this.page.fill(`${type}[name="${name}"]`, value);
+  }
+  async fillByTypeAndLabel(type: string, name: string, value: string) {
+    await this.page.fill(`${type}[label="${name}"]`, value);
```

Comment:
I'm a little confused on when you would use these fill values, do you mind explaining with a comment? 😄

Reply:
`<button class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium ecjmfaa0 css-fenodz" tabindex="0" type="button" data-testid="button" label="Search For Location"><span class="css-rmez2w">Search For Location</span><div class="css-jbvua0"><svg width="14" height="14" xmlns="http://www.w3.org/2000/svg" class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1mkvmnp" focusable="false" aria-hidden="true" viewBox="0 0 14 14" fillcontrast="white"><path d="M6.973 8.977l-5.19-5.328-1.14 1.078 6.328 6.49 6.375-6.423-.886-.963-.184-.081h-.055l-.178.075z" fill-rule="evenodd"></path></svg></div></button>`
This is a rare care I had to use it. existing data-testid="button" added automatically is too generic and one does not know the position of this button on the page. If we manage to a unique id, we will not need this method